src/intersection.class.ts

```diff
@@ -66,24 +65,35 @@ export class Intersection {
    * @param {boolean} [bInfinite=true] check segment intersection by passing `false`
    * @return {Intersection}
    */
-  static intersectLineLine(a1, a2, b1, b2, aInfinite = true, bInfinite = true) {
-    let result;
-    const uaT = (b2.x - b1.x) * (a1.y - b1.y) - (b2.y - b1.y) * (a1.x - b1.x),
-      ubT = (a2.x - a1.x) * (a1.y - b1.y) - (a2.y - a1.y) * (a1.x - b1.x),
-      uB = (b2.y - b1.y) * (a2.x - a1.x) - (b2.x - b1.x) * (a2.y - a1.y);
+  static intersectLineLine(
+    a1: Point,
+    a2: Point,
+    b1: Point,
+    b2: Point,
+    aInfinite = true,
+    bInfinite = true
+  ): Intersection {
+    const b2xb1x = b2.x - b1.x,
+      a1yb1y = a1.y - b1.y,
+      b2yb1y = b2.y - b1.y,
+      a1xb1x = a1.x - b1.x,
+      a2ya1y = a2.y - a1.y,
+      a2xa1x = a2.x - a1.x,
+      uaT = b2xb1x * a1yb1y - b2yb1y * a1xb1x,
+      ubT = a2xa1x * a1yb1y - a2ya1y * a1xb1x,
+      uB = b2yb1y * a2xa1x - b2xb1x * a2ya1y;
```

Comment:
I am guessing all of this can be reduced by using `dotProduct` `crossProduct`
but it is a headache

Don't you prefect subtracting points and working with vectors?

Reply:
I can do that as a fun thing when the rest is done, and only if i save some bytes, otherwise i see it as non useful